operator/pkg/util/merge_iop.go

```diff
@@ -0,0 +1,149 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	yaml2 "github.com/ghodss/yaml"
+	protobuf "github.com/gogo/protobuf/types"
+	v11 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+
+	v1alpha13 "istio.io/api/mesh/v1alpha1"
+	"istio.io/api/networking/v1alpha3"
+	"istio.io/api/operator/v1alpha1"
+	v1alpha12 "istio.io/istio/operator/pkg/apis/istio/v1alpha1"
+)
+
+// Partially mirrored from istio/api and operator/pkg/api (for values).
```

Comment:
Do we need to manually keep this in sync with what is in apis?

Reply:
yes, to some extent. these are only top level structs that aren't likely to change, most fields are not required. structs can be removed from api and remain here without any harm. last of all, we have good test coverage for any potential breaking changes because the merge function is in line with everything.

Reply:
Does it make sense that we can add json tags for the patchStrategy to the api generated by proto?

Reply:
no, this is not possible, because as i mentioned in the comment, the values and meshconfig fields are defined as map[string]interface rather than structs. therefore strategic merge fails with an error if IOP contains any fields for those subtrees that require patchStrategy to be defined.